substrate/utils/wasm-builder/src/lib.rs

```diff
@@ -409,50 +378,58 @@ fn get_bool_environment_variable(name: &str) -> Option<bool> {
 	}
 }
 
-/// Returns whether we need to also compile the standard library when compiling the runtime.
-fn build_std_required() -> bool {
-	let default = runtime_target() == RuntimeTarget::Wasm;
-
-	crate::get_bool_environment_variable(crate::WASM_BUILD_STD).unwrap_or(default)
-}
-
 #[derive(Copy, Clone, PartialEq, Eq)]
 enum RuntimeTarget {
 	Wasm,
 	Riscv,
 }
 
 impl RuntimeTarget {
-	fn rustc_target(self) -> &'static str {
+	/// Creates a new instance.
+	fn new() -> Option<Self> {
+		let Some(value) = env::var_os(RUNTIME_TARGET) else {
+			return Some(Self::Wasm);
+		};
+		if value == "wasm" {
+			Some(Self::Wasm)
+		} else if value == "riscv" {
+			Some(Self::Riscv)
+		} else {
+			None
+		}
+	}
```

Comment:
This should either return a `Result`, or it should do what it was originally doing (a warning + exit). `None` should almost *never* be used to signal an error condition.

Reply:
I'll change this to warning + exit. Changing that semantics is out of scope for the PR.